### PR TITLE
POST /api/technicians/:user_id/reviews

### PR DESCRIPTION
**Pull request by LSB616**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -136,3 +136,76 @@ describe("GET /api/technicians/:user_id", () => {
       });
   });
 });
+
+
+describe.only("POST /api/technicians/:user_id/reviews", () => {
```
Comment (stormytuna) [inline __tests__/app.test.js]: Remove this .only

File: __tests__/app.test.js [hunk 2]
```diff
@@ -136,3 +136,76 @@ describe("GET /api/technicians/:user_id", () => {
...
+  test("status:201, adds new review to the technician object", () => {
+    const newReview = {
+          reviewBody: "This man is a car maniac! 5/7",
+          rating: 4,
+          reviewedBy: 1,
```
Comment (stormytuna) [inline __tests__/app.test.js:146]: reviewedBy should be a string, should technically be an ObjectID but seems like schema is wrong :( 
Set it to someone elses _id for now so it works when we update the schema

File: __tests__/app.test.js [hunk 3]
```diff
@@ -136,3 +136,76 @@ describe("GET /api/technicians/:user_id", () => {
...
+      .expect(201)
+      .then(({ body }) => {
+        const { technician } = body;
+        expect(technician._id).toEqual('63ce75449ae462be0adad72d')
+        expect(technician.technician.reviews).toHaveLength(3);
+        expect(technician.technician.reviews[2]).toMatchObject({
+        reviewBody: "This man is a car maniac! 5/7",
+        rating: "4",
+        reviewedBy: "1",})
+      });
```
Comment (stormytuna) [inline __tests__/app.test.js:160]: Perfect test 👍

File: models/models.technicians.js [hunk 4]
```diff
@@ -28,3 +28,19 @@ exports.findTechnician = async (id) => {
...
+    Object.keys(rest).length > 0 ||
+    typeof reviewBody !== "string" ||
+    typeof rating !== "number" ||
+    typeof reviewedBy !== "number"
+  ) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+
+  await User.findOneAndUpdate({ _id: id }, {$push: {['technician.reviews']: review }})
+  return await User.findById({ _id: id });
```
Comment (stormytuna) [inline models/models.technicians.js:37]: Add a 404 test here, findOneAndUpdate won't return a rejected promise if a user with that id doesnt exist

File: __tests__/app.test.js [hunk 5]
```diff
@@ -136,3 +136,76 @@ describe("GET /api/technicians/:user_id", () => {
...
+        reviewedBy: 15,
+      })
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("Bad request");
+      });
+  });
+});
+
```
Comment (stormytuna) [inline __tests__/app.test.js]: Needs a 404 test for a valid ID but no user with that ID exists

File: __tests__/app.test.js [hunk 6]
```diff
@@ -136,3 +136,76 @@ describe("GET /api/technicians/:user_id", () => {
...
+
+
+
+
+
+
```
Comment (stormytuna) [inline __tests__/app.test.js]: Lots of whitespace here that prettier should be removing

**Review by stormytuna — CHANGES_REQUESTED** [review]

File: app.js [hunk 7]
```diff
@@ -26,10 +26,11 @@ app.get("/api/technicians/:user_id", getTechnician);
 
 app.post("/api/technicians/:user_id/reviews", createReviewforTech)
 
+app.patch("/api/technicians/:user_id", patchTechnician)
 
+app.all("*", handle404s);
```
Comment (stormytuna) [inline app.js:31]: whats this for? `app.use("/*", hand404s)` was fine

File: controllers/controllers.errors.js [hunk 8]
```diff
@@ -3,6 +3,7 @@ exports.handleCustomErrors = (err, req, res, next) => {
     res.status(err.status).send({ msg: err.msg });
   } else {
     next(err);
+    next(err);
```
Comment (stormytuna) [inline controllers/controllers.errors.js]: Probably an artifact from merging, we only need one `next` call here

**Review by stormytuna — COMMENTED** [review]
Couple things to change, fine to merge once they're done

File: app.js (hunk 7 above)
Comment (LSB616) [reply app.js:31]: think it snuck in when I was fixing those merge conflicts... not sure where it's from

File: controllers/controllers.errors.js (hunk 8 above)
Comment (LSB616) [reply controllers/controllers.errors.js]: yeah another merge over sight

**Pull request merged**